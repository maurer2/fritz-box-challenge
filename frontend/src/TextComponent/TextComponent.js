import React, { PureComponent } from 'react';
import styled, { css } from 'styled-components';

const TextWrapper = styled.h1`
  margin: auto;
  color: white;
  /* color: ${props => props.characterCount === 5 ? "green" : "white"}; */
  /*
  ${props => props.numberOfCharacters === 5 && css`
    background: red; 
  `};
  */

  font-size: ${props => 100 / props.characterCount * 1.5 }vw;
`;

class TextComponent extends PureComponent {
  get textSeparate() {
    return this.props.text.split('');
  }

  get numberOfCharacters() {
    return this.textSeparate.length;
  }

  render() {
    return (
      <TextWrapper characterCount={ this.numberOfCharacters }>
        { this.textSeparate.map(
          (character, index) => 
          <span key={ index } > { character } </span>
          )
        }
      </TextWrapper>
    );
  }
}

export default TextComponent;
