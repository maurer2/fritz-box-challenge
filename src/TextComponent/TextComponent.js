import React, { PureComponent } from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

const TextWrapper = styled.h1`
  margin: auto;
  color: white;
  user-select: none;
  /* color: ${props => (props.characterCount === 5 ? 'green' : 'white')}; */
  /*
  ${props => props.numberOfCharacters === 5 && css`
    background: red; 
  `};
  */
  font-size: ${props => 100 / props.characterCount * 1.75}vw;
`;

class TextComponent extends PureComponent {
  get textSeparate() {
    return this.props.text.split('');
  }

  get numberOfCharacters() {
    return this.textSeparate.length;
  }

  render() {
    return (
      <TextWrapper characterCount={ this.numberOfCharacters } className="text-wrapper">
        { this.textSeparate.map(
          (character, index) => (
          <span key={ index }>
            { character }
          </span>
          ),
        )}
      </TextWrapper>
    );
  }
}

TextComponent.propTypes = {
  text: PropTypes.string,
};

export default TextComponent;
