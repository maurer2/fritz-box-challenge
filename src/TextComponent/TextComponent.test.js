import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TextComponent from './TextComponent';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('TextComponent dummy', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<TextComponent text={ '123456' } />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should should contain spans with numbers', () => {
    const wrapper = shallow(<TextComponent text={ '123456' } />);

    expect(wrapper.children('span').length).toBeGreaterThan(0);
    expect(wrapper.children('span').length).toBe(6);
  });

  it('getter should work', () => {
    const wrapper = shallow(<TextComponent text={ '123456' } />);

    const getterNumberOfCharacters = wrapper.instance().numberOfCharacters;
    const { textSeparate } = wrapper.instance();

    expect(getterNumberOfCharacters).toBe(6);
    expect(textSeparate).toEqual(['1', '2', '3', '4', '5', '6']);
  });
});
