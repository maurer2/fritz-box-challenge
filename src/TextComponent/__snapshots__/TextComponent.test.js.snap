// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent dummy should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextComponent
    text="123456"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "characterCount": 6,
      "children": Array [
        <span>
          1
        </span>,
        <span>
          2
        </span>,
        <span>
          3
        </span>,
        <span>
          4
        </span>,
        <span>
          5
        </span>,
        <span>
          6
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "1",
        },
        "ref": null,
        "rendered": "1",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "2",
        },
        "ref": null,
        "rendered": "2",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "3",
        },
        "ref": null,
        "rendered": "3",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "4",
        },
        "ref": null,
        "rendered": "4",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "5",
        },
        "ref": null,
        "rendered": "5",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "6",
        },
        "ref": null,
        "rendered": "6",
        "type": "span",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  margin: auto;
  color: white;
  user-select: none;
  /* color: ",
          [Function],
          "; */
  /*
  ",
          [Function],
          ";
  */
  font-size: ",
          [Function],
          "vw;
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "characterCount": 6,
        "children": Array [
          <span>
            1
          </span>,
          <span>
            2
          </span>,
          <span>
            3
          </span>,
          <span>
            4
          </span>,
          <span>
            5
          </span>,
          <span>
            6
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "1",
          },
          "ref": null,
          "rendered": "1",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "2",
          },
          "ref": null,
          "rendered": "2",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "3",
          },
          "ref": null,
          "rendered": "3",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "4",
          },
          "ref": null,
          "rendered": "4",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "5",
          },
          "ref": null,
          "rendered": "5",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "6",
          },
          "ref": null,
          "rendered": "6",
          "type": "span",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  margin: auto;
  color: white;
  user-select: none;
  /* color: ",
            [Function],
            "; */
  /*
  ",
            [Function],
            ";
  */
  font-size: ",
            [Function],
            "vw;
",
          ],
        },
        "displayName": "styled.h1",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h1",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
